src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,204 @@
+import java.io.BufferedInputStream;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+
+import org.apache.mahout.cf.taste.impl.model.GenericPreference;
+import org.apache.mahout.cf.taste.impl.common.*;
+import org.apache.mahout.cf.taste.impl.model.GenericUserPreferenceArray;
+import org.apache.mahout.cf.taste.impl.model.GenericDataModel;
+import org.apache.mahout.cf.taste.impl.neighborhood.ThresholdUserNeighborhood;
+import org.apache.mahout.cf.taste.impl.recommender.GenericUserBasedRecommender; 
+import org.apache.mahout.cf.taste.model.PreferenceArray;
+import org.apache.mahout.cf.taste.similarity.UserSimilarity;
+import org.apache.mahout.cf.taste.impl.similarity.PearsonCorrelationSimilarity;
+import org.apache.mahout.cf.taste.neighborhood.UserNeighborhood;
+import org.apache.mahout.cf.taste.impl.recommender.GenericRecommendedItem;
+import org.apache.mahout.cf.taste.recommender.RecommendedItem;
+
+
+
+
+
+
+public class MovieRecommender {
+
+    private FastByIDMap<PreferenceArray> usersPreferences; 
+    private HashMap<String,Long> userIDs;
+    private HashMap<String,Long> productIDs;
+    private long userCount;
+    private long productCount;
+    private long reviewCount;
+    private GenericDataModel model;
+    private GenericUserBasedRecommender recommender;
+
+
+    public MovieRecommender(String filePath) {
+
+        productIDs = new HashMap<String,Long>();
+        userIDs = new HashMap<String,Long>();
+        usersPreferences = new FastByIDMap<PreferenceArray>();
+        userCount = 0;
+        productCount =0;
+        reviewCount= 0;
+        
+
+
+        System.out.println("Hello Java Again :(\n");
+        getPreference(filePath);
+        buildRecomender();
+        // printPreferences();
+
+    }
+
+
+    GenericPreference getPreference(String filePath) {
+        String productID = "";
+        String userID = "";
+        Float score = 0.0f;
+
+        BufferedReader reader;
+
+
+        try {
+            reader = new BufferedReader(new FileReader(filePath));
+            String line = reader.readLine();
+            while (line != null) {
+            
+
+                if(line.startsWith("product/productId:") )
+                    productID = line.split(" ")[1];
+                if(line.startsWith("review/userId:") )
+                    userID = line.split(" ")[1];
+                if(line.startsWith("review/score:") ){
+                   
+                    score = Float.parseFloat( line.split(" ")[1]); 
+                    addPreference(userID, productID,score);
+                    
+                    
+                }
+                line = reader.readLine();
+            }
+            reader.close();
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    void addPreference( String userID,String productID, float score)
```

Comment:
Este método hace varias cosas. Podría ser mejor separarlo en métodos más pequeños.

Reply:
Thanks Axel, I created a new function where I handle the insertions in PredefenceArray and ArrayList